editor/scene_tree_dock.cpp

```diff
@@ -2027,27 +2084,18 @@ bool SceneTreeDock::_check_node_path_recursive(Node *p_root_node, Variant &r_var
 				break;
 			}
 
+			if (Object::cast_to<Material>(resource)) {
+				// For performance reasons, assume that Materials don't have NodePaths in them.
+				// TODO This check could be removed when String performance has improved.
+				break;
+			}
```

Comment:
Can't this use the `PROPERTY_HINT_NO_NODEPATH` hint you added instead?

Reply:
No, the problem here is that getting the property list itself is super slow. By the time I get the hint it is already too late.

Reply:
https://github.com/godotengine/godot/blob/master/scene/resources/material.cpp#L2404

This is the function that I need to avoid calling. There's a way to make it a bit faster too, but even then it is still pretty slow.

Reply:
Now I realized this might break ViewportTextures in materials. They won't update paths anymore.

Reply:
They still work, I forgot to put that as a test in my list of things I tested. But that was specifically tested.